pymatgen/entries/computed_entries.py

```diff
@@ -944,11 +1025,22 @@ def from_dict(cls, d) -> "GibbsComputedStructureEntry":
             entry_id=d.get("entry_id", None),
         )
 
-    def __repr__(self):
+    def __repr__(self) -> str:
         output = [
             "GibbsComputedStructureEntry {} - {}".format(
                 self.entry_id, self.composition.formula
             ),
             "Gibbs Free Energy (Formation) = {:.4f}".format(self.energy),
         ]
         return "\n".join(output)
+
+    def __hash__(self) -> int:
+        # NOTE based on hash choice of @mattmcdermott + class name
+        data_md5 = hashlib.md5(
+            f"{self.__class__.__name__}"
+            f"{self.entry_id}"
+            f"{self.composition.reduced_formula}"
+            f"{self._energy}"
+            f"{self.temp}".encode('utf-8')
+        ).hexdigest()
+        return int(data_md5, 16)
```

Comment:
Isn't there an easier way to hash this? e.g. `hash(the_string)`, using hexdigest/hashlib is a bit over complicating I think

Reply:
I adopted `hashlib` to match what @mattmcdermott said he was already using for his `ComputedGibbsStructureEntry`. I think that the python `hash()` isn't guaranteed to be the same across different processes which was a requirement for matt's work. I don't know whether there are alternatives to making `hashlib` a dependency that would keep this functionality.

Reply:
Thanks @CompRhys, that was a subtlety I wasn't aware of, I was under the impression `hash` was fully deterministic for strings. I have no objection to using `hashlib`.